### ISSUE TITLE
`value` parameter considered in scope in `nameof()` in event accessor attribute

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#73733]
**Version Used**:  17.9.7

**Steps to Reproduce**:

Compile the following:
```csharp
using System;

class A : Attribute
{
    public A(string s) { }
}

class C
{
    object P
    {
        [A(nameof(value))] set { }    // error CS0103: The name 'value' does not exist in the current context
    }

    event EventHandler E
    {
        [A(nameof(value))] add { }    // ok
        [A(nameof(value))] remove { } // ok
    }
}
```

**Expected Behavior**:
The same error reported for `[A(nameof(value))]` for event `add` and `remove` as property `set`.

**Actual Behavior**:
No error reported for `[A(nameof(value))]` on event `add` or `remove`.
